openbas-model/src/main/java/io/openbas/database/repository/InjectRepository.java

```diff
@@ -23,6 +23,11 @@ public interface InjectRepository extends CrudRepository<Inject, String>, JpaSpe
   @NotNull
   Optional<Inject> findById(@NotNull String id);
 
+  List<Inject> findByExerciseId(@NotNull String exerciseId);
+
+  @NotNull
+  List<Inject> findByScenarioId(@NotNull String scenarioId);
```

Comment:
Oh that's great
If you have time, you can use it in InjectAPI.scenarioInjects method
It will be clearer